src/main/java/org/opensearch/timeseries/indices/IndexManagement.java

```diff
@@ -1008,6 +1016,53 @@ public <T> void initCustomResultIndexAndExecute(String resultIndexOrAlias, Execu
         }
     }
 
+    /**
+     * creates flattened result index
+     * @param flattenedResultIndexAlias the flattened result index alias
+     * @param actionListener the action listener
+     */
+    public void initFlattenedResultIndex(String flattenedResultIndexAlias, ActionListener<CreateIndexResponse> actionListener) {
+        try {
+            String indexName = getCustomResultIndexPattern(flattenedResultIndexAlias);
+            logger.info("Initializing flattened result index: {}", indexName);
+
+            CreateIndexRequest request = new CreateIndexRequest(indexName)
+                .mapping(getFlattenedResultMappings(), XContentType.JSON)
+                .settings(settings);
+
+            if (flattenedResultIndexAlias != null) {
+                request.alias(new Alias(flattenedResultIndexAlias));
+            }
+
+            choosePrimaryShards(request, false);
+
+            adminClient.indices().create(request, ActionListener.wrap(response -> {
+                if (response.isAcknowledged()) {
+                    logger.info("Successfully created flattened result index: {} with alias: {}", indexName, flattenedResultIndexAlias);
+                    actionListener.onResponse(response);
+                } else {
+                    String errorMsg = "Index creation not acknowledged for index: " + indexName;
+                    logger.error(errorMsg);
+                    actionListener.onFailure(new IllegalStateException(errorMsg));
+                }
+            }, exception -> {
+                logger.error("Failed to create flattened result index: {}", indexName, exception);
+                actionListener.onFailure(exception);
+            }));
+        } catch (Exception e) {
+            logger.error("Error while initializing flattened result index: {}", flattenedResultIndexAlias, e);
+            actionListener.onFailure(e);
+        }
+    }
+
+    public String getFlattenedResultIndexAlias(String indexOrAliasName, String configId) {
+        return indexOrAliasName + "_flattened_" + configId.toLowerCase(Locale.ROOT);
+    }
+
+    public String getFlattenResultIndexIngestPipelineId(String configId) {
+        return "flatten_result_index_ingest_pipeline" + configId.toLowerCase(Locale.ROOT);
+    }
```

Comment:
I would suggest move these to Config and it is a fixed value when creating the config so we don't need to repeatedly reconstruct it.

Reply:
I think you need config id to ensure one flatten index per config. If you use name (which is unique among all config), you can move these methods to config constructor. As these methods are called repeated for every result, I want it to be optimized as much as possible.

Reply:
changed it to use detector name